### PR TITLE
API 800,1000 and 1200 support for Logical enclosure

### PR DESCRIPTION
**Pull request by AnushaY1916**
### Description
API 800,1000 and 1200 support for Logical enclosure

### Issues Resolved
NA

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Review by madhav-bharadwaj — APPROVED** [review]

File: examples/logical_enclosure.pp [hunk 1]
```diff
@@ -13,32 +13,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 ################################################################################
+
```
Comment (sijeesh) [inline examples/logical_enclosure.pp:16]: Let's update the license date in all the modified files

File: lib/puppet/provider/uri_parsing.rb [hunk 2]
```diff
@@ -81,7 +81,8 @@ def special_resources_assign(key)
   when 'associatedUplinkSetUri' then 'UplinkSet'
   when 'parentTaskUri', 'associatedTaskUri' then 'Task'
   when 'nativeNetworkUri' then 'EthernetNetwork'
-  when 'enclosureUris' then 'Enclosure'
+  # Uncomment this line, once array handling is done for enclosureUris
```
Comment (sijeesh) [inline lib/puppet/provider/uri_parsing.rb]: You can add a TODO: and add the issue number.
Ex: TODO: comments with issue number

File: spec/unit/provider/uri_parsing_spec.rb [hunk 3]
```diff
@@ -358,13 +364,14 @@
       expect(uri_validation(data)).to eq 'nativeNetworkUri' => '/rest/ethernet-networks/fake-id'
     end
 
-    it 'should replace Enclosure name by uri when key is enclosureUris' do
-      enclosure = OneviewSDK::API300::C7000::Enclosure.new(@client, 'name' => name, 'uri' => '/rest/enclosures/fake-id')
-      allow(OneviewSDK::API300::C7000::Enclosure).to receive(:find_by).with(anything, name: name).and_return([enclosure])
+    # Use this test case when enclosureUris is handled as Array.
```
Comment (sijeesh) [inline spec/unit/provider/uri_parsing_spec.rb]: Add a TODO comment mentioned as above

File: spec/unit/provider/uri_parsing_spec.rb [hunk 4]
```diff
@@ -380,6 +387,14 @@
       data = { 'osDeploymentPlanUri' => name }
       expect(uri_validation(data)).to eq 'osDeploymentPlanUri' => uri
     end
+
+    it 'should replace SAS Interconnect name by uri when key is SASInterconnectUri' do
```
Comment (sijeesh) [inline spec/unit/provider/uri_parsing_spec.rb]: We can remove this test case as it doesn't cover the intended lines in the uri_parsing module.

**Review by sijeesh — COMMENTED** [review]
Looks good, please check the comments.

File: examples/logical_enclosure.pp (hunk 1 above)
Comment (AnushaY1916) [reply examples/logical_enclosure.pp:16]: Done

File: lib/puppet/provider/uri_parsing.rb (hunk 2 above)
Comment (AnushaY1916) [reply lib/puppet/provider/uri_parsing.rb]: Done

File: spec/unit/provider/uri_parsing_spec.rb (hunk 3 above)
Comment (AnushaY1916) [reply spec/unit/provider/uri_parsing_spec.rb]: Done

File: spec/unit/provider/uri_parsing_spec.rb (hunk 4 above)
Comment (AnushaY1916) [reply spec/unit/provider/uri_parsing_spec.rb]: Removed

**Review by sijeesh — APPROVED** [review]
LGTM

**Pull request merged**